worker/uniter/runner/debug/server_test.go

```diff
@@ -248,6 +248,17 @@ func (s *DebugHooksServerSuite) TestRunHookDebugAt(c *gc.C) {
 	c.Assert(err, jc.ErrorIsNil)
 	c.Check(session.DebugAt(), gc.Equals, "all")
 
+	flockAcquired := make(chan struct{}, 0)
+	waitForFlock := func() {
+		select {
+		case <-flockAcquired:
+		case <-time.After(testing.ShortWait):
+			c.Fatalf("timed out waiting for hook to acquire flock")
+		}
+	}
+	s.PatchValue(&waitClientExit, func(*ServerSession) {
```

Comment:
Where was the patch happening before? I don't see it on the left.

Reply:
We were using the default implementation of `waitClientExit` which spawns a goroutine that ultimately waits for the client script to run. Which meant that the goroutine actually just sat around forever, which was the cause of the race conditions later.